package ProtoNewMatch;

message ProtoHeroProperty {
		required uint32 hid 		= 1;
		required uint32 job 		= 2;
		required uint32 level 		= 3;
		required int32 hp 			= 4;
		repeated uint32 property 	= 5 ;
}

message ProtoNewMatchUser {
		required uint32 uid 		= 1;
		required uint32 score 		= 2;
		required uint32 turn 		= 3;
		repeated uint32 index 		= 4;
		required string name		= 5;
		required string aname		= 6;
		required string fig			= 7;
		repeated ProtoHeroProperty hero  = 8;
}

message ProtoNewMatchGuess {
		required uint32 uid 		= 1;
		required uint32 guess 		= 2;
		required uint32 bet 		= 3;
}
message ProtoNewMatch {
		required uint32 stage 				= 1;
		repeated ProtoNewMatchUser rank		= 2;
		repeated ProtoNewMatchUser top8 	= 3;
		repeated ProtoNewMatchUser top4 	= 4;
		repeated ProtoNewMatchUser top2 	= 5;
		repeated ProtoNewMatchUser user 	= 6;
		repeated ProtoNewMatchGuess guess	= 7;		
}
